api.planx.uk/inviteToPay/createPaymentSendEvents.ts

```diff
@@ -0,0 +1,68 @@
+import { NextFunction, Request, Response } from 'express';
+import { createScheduledEvent } from "../hasura/metadata";
+import { _admin as $admin } from '../client';
+import { getMostRecentPublishedFlow } from '../helpers';
+
+type Destination = "bops" | "uniform" | "email";
+
+interface CombinedResponse {
+  bops?: Record<string, string>;
+  uniform?: Record<string, string>;
+  email?: Record<string, string>;
+}
+
+// Create "One-off Scheduled Events" in Hasura when a payment request is paid
+const createPaymentSendEvents = async (req: Request, res: Response, next: NextFunction): 
+  Promise<NextFunction | Response | void> => {
+  try {
+    const now = new Date();
+    const combinedResponse: CombinedResponse = {};
+
+    // TODO Get this session's published flow data, find it's Send component, determine which "destinations" we need to queue up events for
```

Comment:
This feels more and more to me like a valuable piece of data that's worth exposing on "publish" of a flow maybe?

Reply:
Yes! I agree storing this on the table makes sense. We can populate/update on the publishing step? 

Was thinking about this and also realised another on-publish validation step we'll want: currently, when Invite to Pay is enabled via the Pay component, we check for the existence of a Send component. But, I think we actually need to check for _exactly one_ Send component. 

Otherwise, the flow could have many Send components with different destinations selected each on branches. This is fine if it's a non-Invite to Pay flow where we know all applicants stay in the graph, but I don't think we want to have to calculate which branched Send/destinations the Invite to Pay applicant _should_ have reached when queuing up these payment request events (eg API has no knowledge of upcoming node ids). 

It doesn't seem like more than one Send is a pattern that shows up in flow content anyways right now so hopefully won't ever actually be an issue, but adding a validation step will make sure we know if/when that changes!

Reply:
`findNextNodeOfType` in [planx-core](https://github.com/theopensystemslab/planx-core/blob/main/src/models/session/logic.ts#L104) is able to find the right Pay component using the flow and breadcrumbs. It should work for Send too.

Reply:
In this case, there won't be a Send component in my `breadcrumbs` though - so I wrote a simple `.find()` instead that just iterates over a flattened flow looking for a match on component type & extracts it's `destinations` from node data.

I've also updated the invite to pay validation rules to be a stricter: rather than just checking for the existance of any Send component(s), I'm now checking that there's exactly one. That'll save us having to collate possibly many Send components that each set different "destinations" like explained above. I don't think this will invalidate any existing flow content.

Reply:
I think `findNextNodeOfType` does what you wanted to do - it looks for the next component of the type on the flow (naively at the moment, but it should evolve to handling the edge cases of traversing edges properly soon). It's using breadcrumbs to find what's upcoming but it's checking the flow to find the upcoming nodes.
Since it's not robust enough to handle every edge case yet, restricting editor functionality for the time-being makes sense but I'd be keen to come back to remove these kinds of limitations where possible.